spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/AbstractMqttMessageDrivenChannelAdapter.java

```diff
@@ -54,14 +55,35 @@
 
 	protected final Lock topicLock = new ReentrantLock(); // NOSONAR
 
-	public AbstractMqttMessageDrivenChannelAdapter(String url, String clientId, String... topic) {
+	/**
+	 * Construct an instance with the provided properties.
+	 * @param clientId the client id.
+	 * @param topics the topics.
+	 * @since 5.4.
+	 */
+	@SuppressWarnings("deprecation")
+	public AbstractMqttMessageDrivenChannelAdapter(String clientId, String... topics) {
+		this(null, clientId, topics);
+	}
+
+	/**
+	 * Deprecated - do not use; implementations must maintain the url or other connection
+	 * information.
+	 * @deprecated in favor of
+	 * {@link #AbstractMqttMessageDrivenChannelAdapter(String, String...)}.
+	 * @param url the url.
+	 * @param clientId the client id.
+	 * @param topics the topics.
+	 */
+	@Deprecated
```

Comment:
Yeah... I didn't mean to deprecate per-channel-adapter URL configuration.
My idea was just to deprecate a `getUrl()` since it not always reflect a connection reality.

Since we have now that `MqttConnectOptions getConnectionInfo()` may be it is better to clone its instance and modify its `serverURIs` with the provided URL. So, we still can keep existing apps without respective modification for per-channel-adapter-client-factory.

WDYT?

Reply:
Good point - I hadn't thought of the shared factory.